### PR TITLE
Russell/trackerpopup

### PR DESCRIPTION
**Pull request by russellholt**
To understand @laurengarcia 's PR lauren/build-upgrade, I implemented a first shot at the main popup using test data (until the company/stats module is finished). The purpose of this PR is discussion. Ideally we use it to talk about the best way to organize the views, accessing data through the models, etc. This might be closed or if it's on the right track we could just go ahead and merge the two, and work from there. either way..

I did some css to make it roughly similar to the mockup, and to verify how to reference images, etc in the extension, from the new tagged templates.

- main popup switched to html/trackers.html
- four new views:
    1. search box
    2. site info (rating, etc)
    3. top tracker list
    4. linked menu item (arrow)

screen shot 2017-05-02 at 12 41 38 am

**Reviewer:**

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: js/templates/trackerlist.es6.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+module.exports = function () {
+
+    var f = function(list) {
+        return list.map((site) => bel`<li> ${site.domain} = ${site.blocked} </li>`);
+    };
+
+    return bel`<div class="js-trackerlist js-menu-section">
+        <div class="js-menu-title">${this.model.heading}</div>
+        <ul class="js-menu-item-list">
+            ${f(this.model.testList)}
```
Comment (laurengarcia) [inline js/templates/trackerlist.es6.js]: This looks great. Only criticism is that I generally lean away from single variable function names in favor of something more descriptive. So maybe `f()` becomes `printList()` or something.

File: js/views/search.es6.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    // this.bindEvents() wires up jQuery selectors to events and their handlers:
+    this.bindEvents([
+      [this.$go, 'click', this._handleSubmit]
+    ]);
+
+    this.bindEvents([
+      [this.$form, 'submit', this._handleSubmit]
+    ]);
+
```
Comment (laurengarcia) [inline js/views/search.es6.js]: This view looks fantastic. Only thing: `this.bindEvents` takes an array of arrays as its argument, so you can combine your two calls like so:
`this.bindEvents([ [this.$go... etc], [this.$form... etc] ])`
https://dub.duckduckgo.com/core/ddg/blob/bttf/www-release/frontend/src/js/views/Ads.js#L25

File: js/views/site.es6.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Parent.call(this, ops);
+
+    console.log("new site view");
+
+
+    // this._cacheElems() caches jQuery selectors, so the following would be
+    // accessible via: `this.$item` from within this view
+    // and is equivalent to $('.js-site-item')
+
+    this._cacheElems('.js-site', [ 'whitelistToggle' ]);
```
Comment (laurengarcia) [inline js/views/site.es6.js]: please use `kebab-case` vs `camelCase` for class names. So `whitelist-toggle` vs what's there.

File: css/trackers.scss [hunk 4]
```diff
@@ -0,0 +1,177 @@
+@import "../node_modules/duckduckgo-colors/colors";
+
+body {
+    margin:0;
+    padding:0;
+    font-size:14px;
+    font-weight:bolder;
+}
```
Comment (laurengarcia) [inline css/trackers.scss]: We should probably talk about css cascade at some point. Ideally, general rules about the `body` elem should go way higher in the cascade, probably in a `normalize.scss` file and then a `base.scss` file. By the time you get down into something specific like `trackers.scss`, there shouldn't be a need for this here.

File: css/trackers.scss [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+#DDG-site-info {
```
Comment (laurengarcia) [inline css/trackers.scss]: let's have all classes and id's be lower case.

File: js/pages/trackers.es6.js [hunk 6]
```diff
@@ -1,27 +1,72 @@
...
+const SiteModel = require('./../models/site.es6.js');
+const SiteTemplate = require('./../templates/site.es6.js');
+
+const SearchView = require('./../views/search.es6.js');
+const SearchModel = require('./../models/search.es6.js');
+const SearchTemplate = require('./../templates/search.es6.js');
+
+const ItemMenuView = require('./../views/itemMenu.es6.js');
+const ItemMenuModel = require('./../models/itemMenu.es6.js');
+const ItemMenuTemplate = require('./../templates/itemMenu.es6.js');
```
Comment (laurengarcia) [inline js/pages/trackers.es6.js]: Let's declare templates like `itemMenuTemplate` vs `ItemMenuTemplate`. Generally uppercase first letter is used to declare constructors. Our templates are just plain old functions. It's pretty common in the SERP codebase that functions are all uppercase, but I feel that this is confusing/misleading.

File: js/templates/trackerlist.es6.js (hunk 1 above)
Comment (russellholt) [inline js/templates/trackerlist.es6.js]: totally agree!

File: js/views/search.es6.js (hunk 2 above)
Comment (russellholt) [inline js/views/search.es6.js]: Cool, that makes sense. one thing though, don't want to link there from this external repo.

File: css/trackers.scss (hunk 4 above)
Comment (russellholt) [inline css/trackers.scss]: for sure. I'm kind of in the backhoe/bulldozer phase here :)

File: js/views/itemMenu.es6.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    this.pageView = ops.pageView;
+    this.template = ops.template;
+
+    Parent.call(this, ops);
+
+    console.log("new itemMenu view");
+
+    // this._cacheElems('#js-item-menu', [ this.model.id ]);
+
+    this.$linkableItem = $("#js-item-menu-" + this.model.id);
```
Comment (russellholt) [inline js/views/itemMenu.es6.js]: @laurengarcia  better way to do this?

**Comment by russellholt** [comment]
@laurengarcia addressed most of these except the itemMenuTemplate naming.

**Pull request merged**